internal/bminventory/inventory.go

```diff
@@ -1581,24 +1582,35 @@ func calculateHostNetworks(log logrus.FieldLogger, cluster *common.Cluster) []*m
 
 func (b *bareMetalInventory) ListClusters(ctx context.Context, params installer.ListClustersParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
-	var clusters []*common.Cluster
-	if err := b.db.Preload("Hosts").Where(identity.AddUserFilter(ctx, "")).Find(&clusters).Error; err != nil {
-		log.WithError(err).Error("failed to list clusters")
-		return installer.NewListClustersInternalServerError().
-			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
-	}
-	var mClusters []*models.Cluster = make([]*models.Cluster, len(clusters))
-	for i, c := range clusters {
-		mClusters[i] = &c.Cluster
+	db := b.db
+	if *params.GetUnregisteredClusters {
+		db = db.Unscoped().Where("deleted_at > ?", strfmt.DateTime{})
+	}
```

Comment:
I imagined the semantics would be that if this parameter was set we would get ALL the clusters (registered and unregisted), not only unregistered.
@ronniel1 @filanov thought?

Reply:
I agree